doc/nrf/libraries/bin/lwm2m_carrier/CHANGELOG.rst

```diff
@@ -9,15 +9,55 @@ Changelog
 
 All notable changes to this project are documented in this file.
 
-liblwm2m_carrier 3.3.3
+Certification status
+====================
+
+For certification status of the released versions, see `Mobile network operator certifications`_.
+
+liblwm2m_carrier 3.4.0
 **********************
 
-Release for modem firmware version 1.3.5 and 2.0.0.
+Release for modem firmware version 1.3.5, 1.3.6, 2.0.0, and 2.0.1.
 
-Certification status
-====================
+Size
+====
 
-For certification status, see `Mobile network operator certifications`_.
+See :ref:`lwm2m_lib_size` for an explanation of the library size in different scenarios.
+
++-------------------------+---------------+------------+
+|                         | Flash (Bytes) | RAM (Bytes)|
++-------------------------+---------------+------------+
+| Library size            |               |            |
+| (binary)                |               |            |
++-------------------------+---------------+------------+
+| Library size            |               |            |
+| (reference application) |               |            |
++-------------------------+---------------+------------+
+
+Changes
+=======
+
+* Removed AT&T support. The LwM2M carrier library is no longer required to certify with AT&T.
+
+* Added preliminary support for Bell Canada subscriber ID.
+  This carrier can be disabled or enabled with the Kconfig option :kconfig:option:`CONFIG_LWM2M_CARRIER_BELL_CA`.
+
+* Changed the default string of the Device Type resource to say "Module" instead of "Smart Device".
+  This can be changed to other strings using :kconfig:option:`CONFIG_LWM2M_CARRIER_DEVICE_TYPE`.
+
+* Added a timeout to abort Push FOTA operations :kconfig:option:`CONFIG_LWM2M_CARRIER_FIRMWARE_DOWNLOAD_TIMEOUT`.
+  By default (0) the timer is disabled for unknown Subscriber IDs, and set to 30 minutes by for the SoftBank Subscriber ID.
+
+* Removed :kconfig:option:`LWM2M_CARRIER_THREAD_STACK_SIZE`, and the corresponding thread from the glue layer.
+
+* Added the function :c:func:`lwm2m_carrier_data_send`. This function can be used to send Binary App Data Container and Event Log object data.
+
+  * Renamed the old ``lwm2m_carrier_app_data_send`` to func:`lwm2m_carrier_app_data_set` to avoid confusion with the new :c:func:`lwm2m_carrier_data_send` function. The name now also matches the similar :c:func:`lwm2m_carrier_log_data_set`.
```

Comment:
```suggestion
  * Renamed the old ``lwm2m_carrier_app_data_send`` to :c:func:`lwm2m_carrier_app_data_set` to avoid confusion with the new :c:func:`lwm2m_carrier_data_send` function.
    The name now also matches the similar function :c:func:`lwm2m_carrier_log_data_set`.
```